src/Microsoft.Extensions.Primitives/CancellationChangeToken.cs

```diff
@@ -31,15 +31,31 @@ public CancellationChangeToken(CancellationToken cancellationToken)
         /// <inheritdoc />
         public IDisposable RegisterChangeCallback(Action<object> callback, object state)
         {
+            // Don't capture the current ExecutionContext and its AsyncLocals onto the token registration causing them to live forever
+            var restoreFlow = false;
             try
             {
+                if (!ExecutionContext.IsFlowSuppressed())
```

Comment:
Should this outside the `if` block? Doesn't really need a `try-catch` does it?

Reply:
is the `try` `finally` its for, the `catch` was there previously

Reply:
Moved `if` outside